src/main/java/org/dataone/hashstore/hashfs/HashFileStore.java

```diff
@@ -127,91 +121,124 @@ public HashAddress putObject(InputStream object, String pid, String additionalAl
      * @return A HashAddress object that contains the file id, relative path,
      *         absolute path, duplicate status and a checksum map based on the
      *         default algorithm list.
-     * @throws IOException
-     * @throws NoSuchAlgorithmException
-     * @throws SecurityException
-     * @throws FileNotFoundException
-     * @throws FileAlreadyExistsException
-     * @throws IllegalArgumentException
-     * @throws NullPointerException
+     * @throws IOException                I/O Error when writing file, generating
+     *                                    checksums and moving file or deleting
+     *                                    tmpFile upon duplicate found
+     * @throws NoSuchAlgorithmException   When additiionalAlgorithm or
+     *                                    checksumAlgorithm is invalid
+     * @throws SecurityException          Insufficient permissions to read/access
+     *                                    files or when generating/writing to a file
+     * @throws FileNotFoundException      When file tmpFile not found during store
+     * @throws FileAlreadyExistsException Duplicate object in store exists
+     * @throws IllegalArgumentException   When signature values are unexpectedly
+     *                                    empty (checksum, pid, etc.)
+     * @throws NullPointerException       Arguments are null for pid or object
      */
-    protected HashAddress put(InputStream object, String pid, String additionalAlgorithm, String checksum,
+    public HashAddress putObject(InputStream object, String pid, String additionalAlgorithm, String checksum,
             String checksumAlgorithm)
             throws IOException, NoSuchAlgorithmException, SecurityException, FileNotFoundException,
             FileAlreadyExistsException, IllegalArgumentException, NullPointerException {
+        logHashFileStore.info("HashFileStore.putObject - Called to put object for pid: " + pid);
+        // Begin input validation
         if (object == null) {
+            logHashFileStore.error("HashFileStore.putObject - InputStream cannot be null, pid: " + pid);
             throw new NullPointerException("Invalid input stream, data is null.");
         }
         // pid cannot be empty or null
-        if (pid == null || pid.isEmpty()) {
-            // TODO: Log failure - include signature values
+        if (pid == null || pid.trim().isEmpty()) {
+            logHashFileStore.error("HashFileStore.putObject - pid cannot be null or empty. pid: " + pid);
             throw new IllegalArgumentException("The pid cannot be null or empty");
         }
-
         // Checksum cannot be empty or null if checksumAlgorithm is passed
         if (checksumAlgorithm != null & checksum != null) {
```

Comment:
This if statement doesn't catch the scenario that checksum is null.

Reply:
Resolved when I fixed the same issue in HashStore